### ISSUE TITLE
Regent does not typecheck a region correctly if its fspace has an array

### ISSUE THREAD

**Issue by sparkasaurusRex** [StanfordLegion/legion#667]
I believe a recent commit broke my Regent project. I have a region of type `double[5]` and I'm suddenly getting this error.
```
legion/language/src/regent/std.t:1014: Errors reported during typechecking.
top.rg:68: type mismatch in argument 1: expected region(ispace(int2d), double[5]) but got region(ispace(int2d), double[5])
  populateGammaTable(r_gamma_table)
```

Here is an example.

```
import "regent"

task foo(r : region(ispace(int1d), double[5]))
where
  reads writes(r)
do
  for index in r do
    r[index][0] = 1.2
  end
end

task toplevel()
  var r = region(ispace(int1d, 5), double[5])
  foo(r)
end
regentlib.start(toplevel)
```

And the error.
```
legion/language/src/regent/std.t:1014: Errors reported during typechecking.
test.rg:15: type mismatch in argument 1: expected region(ispace(int1d), double[5]) but got region(ispace(int1d), double[5])
  foo(r)
     ^

stack traceback:
	[C]: in function 'error'
	legion/language/terra/src/terralib.lua:388: in function 'finishandabortiferrors'
	legion/language/src/common/report.t:56: in function 'error'
	legion/language/src/regent/std.t:1014: in function 'validate_args'
	legion/language/src/regent/type_check.t:967: in function 'expr'
	legion/language/src/regent/type_check.t:4247: in function 'fn'
	legion/language/terra/src/terralist.lua:150: in function 'map'
	legion/language/src/regent/type_check.t:3859: in function 'block'
	legion/language/src/regent/type_check.t:4587: in function <legion/language/src/regent/type_check.t:4520>
	legion/language/src/regent/passes.t:69: in function <legion/language/src/regent/passes.t:57>
	test.rg:12: in main chunk
```

I think the error also happens when `double[]` is a part of the fspace.

**Comment by sparkasaurusRex** [comment]
Tested a random commit in the past and I think 0bd0b591ea33a335f0ea169a14125d9bd85e0ad7 works fine, so I'll stay on that for now.

**Comment by magnatelee** [comment]
This should be fixed as of 482f7e65. @sparkasaurusRex Can you pull and try again?

**Comment by sparkasaurusRex** [comment]
Fixed, thanks!
